node/flatpak-node-generator.py

```diff
@@ -619,11 +652,51 @@ def process_lockfile(self, lockfile: Path) -> Iterator[Package]:
         raise NotImplementedError()
 
 
+class RCFileProvider:
+    def get_node_headers(self, rcfile: Path) -> NodeHeaders:
+        raise NotImplementedError()
+
+
 class ModuleProvider(contextlib.AbstractContextManager):
     async def generate_package(self, package: Package) -> None:
         raise NotImplementedError()
 
 
+class NodeHeadersProvider(contextlib.AbstractContextManager):
+    def __init__(self, gen: ManifestGenerator):
+        self.gen = gen
+
+    def __exit__(self, *_: Any) -> None:
+        pass
+
+    def get_gyp_dir(self) -> Path:
+        return self.gen.data_root / 'cache' / 'node-gyp'
+
+    async def generate_node_headers(self, node_headers: NodeHeaders, dest: Path = None):
+        url = node_headers.get_url()
+        install_version = node_headers.get_install_version()
+        if dest is None:
+            dest = self.get_gyp_dir() / node_headers.target
+        metadata = await RemoteUrlMetadata.get(url, cachable=True)
+        self.gen.add_archive_source(url,
+                                    metadata.integrity,
+                                    destination=dest)
+        self.gen.add_data_source(install_version,
+                                 destination=dest / 'installVersion')
+
+    async def generate_sdk_node_headers(self):
+        gyp_dir = self.get_gyp_dir()
+        install_version = "9"
+        script = f'''version="$(node --version | sed \'s/^v//\')"
+                     nodedir="$(dirname $(dirname $(which node)))"
```

Comment:
Will this work if `enable.sh` is used? In that case, `node` wouldn't be in the path yet when the `shell` source commands are run... (tbh you could probably get away with just globbing for `/usr/lib/sdk/node*`, I'm not aware of any packages that actually need *multiple* versions of node at once)

Reply:
The problem is that all installed SDK extensions are always added to sandbox. So, for local builds, multiple `node`'s are likely to be there.

Reply:
That's definitely an oof. I don't really want to break enable.sh though, at least while it's still there, at minimum it should then be removed to avoid bizarre behavior. (Ideally we'd keep it since it's nice for debugging but...I legit have no idea how to handle this then. :neutral_face: ) Would it be *terrible* to go through all the directories since they'll have different versions anyway?

Reply:
I guess going through all the directories has 50/50 chance to pick up wrong version. Maybe add a CLI option to set major node version explicitly, e.g. `--node-sdk-ext=node10`?